src/org/usfirst/frc/team467/robot/Elevator.java

```diff
@@ -0,0 +1,163 @@
+	 * New max: 2980, new min:956.5
+	 * 16.9 ticks = 1 inch
+	 * 1 rotation=253 ticks
+	 * 
+	 * @return a single instance of the Elevator object.
+	 */
+	public static Elevator getInstance() {
+		if (instance == null) {
+			instance = new Elevator();
+		}
+		return instance;
+	}
+
+	/**
+	 * Moves based on the Xbox controller analog input
+	 * 
+	 * @param speed The velocity. Shall be a value between -1 and 1.
+	 */
+	public void manualMove(double speed) {
+
+		targetHeight = Stops.noStop;
+		if (m_safetyHelper != null) {
+			m_safetyHelper.feed();
+		}
+		if (isOutOfRange()) {
+			heightController.set(0);
+			//DriverStation.getInstance().setDriverRumble(0.5);
+			return; // Don't bother with any more logic here.
+		}
+		
+		double currentHeight = getHeight();
+		for (Stops stop : Stops.values()) {
+			if ((previousHeight < stop.height && currentHeight >= stop.height)
+				|| (previousHeight > stop.height && currentHeight <= stop.height)) {
+				
+				//DriverStation.getInstance().setDriverRumble(0.5);
+			} else {
+				//DriverStation.getInstance().setDriverRumble(0.0);
+			}
+		}
+		if(currentHeight >= 2 && currentHeight <= 2.1 || currentHeight >= 4 && currentHeight <=4.1 || currentHeight >= 6 && currentHeight <= 6.1)
+			LOGGER.debug("Rumbling right now");
+			else {		
+			}
+		LOGGER.debug("Current Height: " + currentHeight);
+		heightController.set(speed);
+		previousHeight = currentHeight;
+		LOGGER.debug("Previous Height: " + previousHeight);
+	}
+	
+	public void initMotionMagicMode() {
+//		heightController.setSelectedSensorPosition(0, 0, RobotMap.TALON_TIMEOUT);
+
+		
+		double kPElevator = 1.4; // Double.parseDouble(SmartDashboard.getString("DB/String 7", "1.4"));
+		double kIElevator = 0.0; // Double.parseDouble(SmartDashboard.getString("DB/String 8", "0.0"));
+		double kDElevator = 165; // Double.parseDouble(SmartDashboard.getString("DB/String 9", "165"));
+		double kFElevator = 0.5; //  Double.parseDouble(SmartDashboard.getString("DB/String 6", "0.5"));
+		
+		heightController.config_kP(0, kPElevator, RobotMap.TALON_TIMEOUT);
+		heightController.config_kI(0, kIElevator, RobotMap.TALON_TIMEOUT);
+		heightController.config_kD(0, kDElevator, RobotMap.TALON_TIMEOUT);
+		heightController.config_kF(0, kFElevator, RobotMap.TALON_TIMEOUT);
+
+		//		This is commented out because we will need the SmartDashboard to tune other things later.
+		
+		heightController.configMotionCruiseVelocity(maxTicksPerIteration / 2, RobotMap.TALON_TIMEOUT);
+		heightController.configMotionAcceleration(maxTicksPerIteration / 2, RobotMap.TALON_TIMEOUT);	
+	}
+
+	public boolean isOutOfRange() {
+		return (getHeight() > RobotMap.ELEVATOR_MAX_HEIGHT_IN_FEET || getHeight() < RobotMap.ELEVATOR_MIN_HEIGHT_IN_FEET);
+	}
+
+	public double getHeight() {
+		double height = (heightSensor.getValue() - RobotMap.ELEVATOR_INITIAL_TICKS) * feetPerTick;
+		LOGGER.debug("Height in feet: " + height);
+		return height;
+	}
+	
+	public void targetHeight(Stops target) {
```

Comment:
I think all of the "height" related methods should include the units in their name. Otherwise you could confuse feet with inches and crash your probe into the surface of the planet.